### PR TITLE
[AArch64] Cleanup smashable instrs.

### PR DESCRIPTION
**Pull request by dave-estes**
Changes include a more robust implementation of the smashable
target/immediate lookup code using VIXL helper functions instead
of simpler, hard-coded checks. Additionally a few VIXL helper
functions were added. Lastly, the service request constants were
updated to accommodate their use of smashables.

### CODE REVIEW

**Comment by dave-estes** [comment]
Hey @mxw, I split out the smashable-related code from #7661. Will comment there as well as I address the rest of those comments.

**Comment by mxw** [comment]
This looks good to me.  What test suite are you guys using, by the way?

cc/ @cmuellner, @jim-saxman, @swalk-cavium

**Comment by dave-estes** [comment]
@mxw, we're running test/all and OSS. I typically don't run all combinations of the options for test/all, but I know @jim-saxman does.

**Comment by mxw** [comment]
I'm going to try to land this today.

File: hphp/runtime/vm/jit/smashable-instr-arm.cpp [hunk 1]
```diff
@@ -198,7 +198,20 @@ void smashJcc(TCA inst, TCA target, ConditionCode cc) {
 ///////////////////////////////////////////////////////////////////////////////
 
 uint64_t smashableMovqImm(TCA inst) {
-  return *reinterpret_cast<uint64_t*>(inst + smashableMovqLen() - 8);
+  using namespace vixl;
+
+  Instruction* ldr = Instruction::Cast(inst);
+  Instruction* b = ldr->NextInstruction();
+  Instruction* target = b->NextInstruction();
+  Instruction* after = target->NextInstruction()->NextInstruction();
```
Comment (mxw) [inline hphp/runtime/vm/jit/smashable-instr-arm.cpp]: Everything that's only used in the assertx needs to be declared `DEBUG_ONLY`, e.g.

```
DEBUG_ONLY auto const ldr = Instruction::Cast(inst);
...
```

File: hphp/runtime/vm/jit/smashable-instr-arm.cpp [hunk 2]
```diff
@@ -208,57 +221,83 @@ uint32_t smashableCmpqImm(TCA inst) {
...
-  if (b->Bits(31, 24) != 0x54 || b->Bit(4) != 0) return CC_NP;
 
-  Instruction* br = Instruction::Cast(inst + 8);
-  if (br->Bits(31, 10) != 0x3587C0 || br->Bits(4, 0) != 0) return CC_NP;
+  Instruction* b = Instruction::Cast(inst);
+  Instruction* ldr = b->NextInstruction();;
+  Instruction* br = ldr->NextInstruction();
+  Instruction* target = br->NextInstruction();
+  Instruction* after = target->NextInstruction()->NextInstruction();
+  const auto rd = ldr->Rd();
```
Comment (mxw) [inline hphp/runtime/vm/jit/smashable-instr-arm.cpp]: Here also.

**Pull request closed without merging**